mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -106,74 +105,71 @@ public void update(CRUDOperationContext ctx,
         int numMatched = 0;
         int numFailed = 0;
         BsonMerge merge = new BsonMerge(md);
+        BulkUpdateCtx bulkCtx=new BulkUpdateCtx(collection);
         try {
             ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_RESULTSET, ctx);
             cursor = collection.find(query, null);
```

Comment:
What would it take to be able to catch concurrent mods and retry from this point, constraining the retry-find to only those document ids which had concurrent mods? To me this would be a killer feature (essentially it would make very expressive updates appear atomic), but I realize maybe hard to implement. Is this something worth considering, if only long term? Or would it be a bad idea?